arch/cpu/gecko/Makefile.gecko

```diff
@@ -114,16 +121,40 @@ GSDK_INC_PATHS += $(GSDK_ROOT)/platform/service/iostream/inc
 
 ### device_init
 GSDK_C_SRCS += $(GSDK_ROOT)/platform/service/device_init/src/sl_device_init_nvic.c
+ifeq ($(SL_SERIES),1)
 GSDK_C_SRCS += $(GSDK_ROOT)/platform/service/device_init/src/sl_device_init_dcdc_s1.c
```

Comment:
Replace the `_s1.c` with `_s$(SL_SERIES).c` instead of duplicating the entire block.